src/usr/local/www/vpn_ipsec_settings.php

```diff
@@ -68,6 +70,33 @@
 		}
 	}
 
+	$bypassrules = array();
+	for ($x = 0; $x < 99; $x++) {
+		if (isset($_POST["source{$x}"]) && isset($_POST["destination{$x}"]) &&
+		    isset($_POST["srcmask{$x}"]) && isset($_POST["dstmask{$x}"])) {
+			$source = htmlspecialchars($_POST["source{$x}"]) . '/' . htmlspecialchars($_POST["srcmask{$x}"]);
+			$destination = htmlspecialchars($_POST["destination{$x}"]) . '/' . htmlspecialchars($_POST["dstmask{$x}"]);
+			$rule = array();
+			$rule['source'] = htmlspecialchars($_POST["source{$x}"]);
+			$rule['srcmask'] = htmlspecialchars($_POST["srcmask{$x}"]);
+			$rule['destination'] = htmlspecialchars($_POST["destination{$x}"]);
+			$rule['dstmask'] = htmlspecialchars($_POST["dstmask{$x}"]);
```

Comment:
1. The validation should happen before the values are stored.
2. Why run all of these through `htmlspecialchars()` here? If they get printed in the error, run that through `htmlspecialchars()` if they fail validation before using them in the error message. If they pass validation they wouldn't need encoding since we are certain they are valid IP addresses/subnets.